File: app/src/main/java/com/asksira/loopingviewpagerdemo/DemoInfiniteAdapter.java
package com.asksira.loopingviewpagerdemo;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.widget.TextView;

import com.asksira.loopingviewpager.LoopingPagerAdapter;

import java.util.ArrayList;

public class DemoInfiniteAdapter extends LoopingPagerAdapter<Integer> {

    private static final int VIEW_TYPE_NORMAL = 100;
    private static final int VIEW_TYPE_SPECIAL = 101;

    public DemoInfiniteAdapter(Context context, ArrayList<Integer> itemList, boolean isInfinite) {
        super(context, itemList, isInfinite);
    }

    @Override
    protected int getItemViewType(int listPosition) {
        if (itemList.get(listPosition) == 0) return VIEW_TYPE_SPECIAL;
        return VIEW_TYPE_NORMAL;
    }

    @Override
    protected View inflateView(int viewType, int listPosition) {
        if (viewType == VIEW_TYPE_SPECIAL) return LayoutInflater.from(context).inflate(R.layout.item_special, null);
        return LayoutInflater.from(context).inflate(R.layout.item_pager, null);
    }

    @Override
    protected void bindView(View convertView, int listPosition, int viewType) {
        if (viewType == VIEW_TYPE_SPECIAL) return;
        convertView.findViewById(R.id.image).setBackgroundColor(context.getResources().getColor(getBackgroundColor(listPosition)));
        TextView description = convertView.findViewById(R.id.description);
        description.setText(String.valueOf(itemList.get(listPosition)));
    }


    private int getBackgroundColor (int number) {
        switch (number) {
            case 0:
                return android.R.color.holo_red_light;
            case 1:
                return android.R.color.holo_orange_light;
            case 2:
                return android.R.color.holo_green_light;
            case 3:
                return android.R.color.holo_blue_light;
            case 4:
                return android.R.color.holo_purple;
            case 5:
                return android.R.color.black;
            default:
                return android.R.color.black;
        }
    }
}

File: app/src/main/java/com/asksira/loopingviewpagerdemo/DemoInfiniteAdapter.kt
package com.asksira.loopingviewpagerdemo

import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.viewbinding.ViewBinding
import com.asksira.loopingviewpager.LoopingPagerAdapter
import com.asksira.loopingviewpagerdemo.databinding.ItemPageBinding
import com.asksira.loopingviewpagerdemo.databinding.ItemSpecialBinding
import java.util.*

class DemoInfiniteAdapter(
    itemList: ArrayList<Int>,
    isInfinite: Boolean
) : LoopingPagerAdapter<Int>(itemList, isInfinite) {

    override fun getItemViewType(listPosition: Int): Int {
        return if (itemList?.get(listPosition) == 0) VIEW_TYPE_SPECIAL else VIEW_TYPE_NORMAL
    }

    override fun inflateView(viewType: Int, container: ViewGroup, listPosition: Int): ViewBinding {
        return if (viewType == VIEW_TYPE_SPECIAL) {
            ItemSpecialBinding.inflate(
                LayoutInflater.from(
                    container.context
                ), container, false
            )
        } else {
            ItemPageBinding.inflate(
                LayoutInflater.from(
                    container.context
                ), container, false
            )
        }
    }

    override fun bindView(
        binding: ViewBinding,
        listPosition: Int,
        viewType: Int
    ) {
        if (binding is ItemSpecialBinding) return
        if (binding is ItemPageBinding) {
            binding.image
                .setBackgroundColor(
                    binding.root.context.resources.getColor(
                        getBackgroundColor()
                    )
                )
            binding.message = itemList?.get(listPosition).toString()
        }
    }

    val colorSelector = mutableListOf<Int>(
        android.R.color.holo_red_light,
        android.R.color.holo_orange_light,
        android.R.color.holo_green_light,
        android.R.color.holo_blue_light,
        android.R.color.holo_purple
    )

    private fun getBackgroundColor(): Int {
        return colorSelector.random()
    }

    companion object {
        private const val VIEW_TYPE_NORMAL = 100
        private const val VIEW_TYPE_SPECIAL = 101
    }
}